x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/indexing/AsyncTwoPhaseIndexer.java

```diff
@@ -154,7 +154,11 @@ public synchronized boolean maybeTriggerAsyncJob(long now) {
                 // fire off the search. Note this is async, the method will return from here
                 executor.execute(() -> {
                     onStart(now, ActionListener.wrap(r -> {
-                        nextSearch(ActionListener.wrap(this::onSearchResponse, this::finishWithSearchFailure));
+                        if (r == null || r) {
```

Comment:
Just out of curiosity, is there a reason the response will ever be null instead of true/false?  Should we assert or throw or something in that case?

Edit: I see below some places return `null` instead of `true`.  Does that have significance then?

Reply:
@polyfractal I wanted the "default" action to be `true` here. Hence the `null` check to just be considered `true`. I did change all the interactions (I think) to return `true` instead of `null`. 

We could assert here and allow the `onFailure` handler take care of the rest? I do not have a strong opinion.

Reply:
No strong opinion here either, it just seemed like `null` would be closer to a failure or undefined/unexpected situation.  

If we leave as-is I'd just drop a comment explaining why ("default to be true/run"), or maybe the failure handler handles `r == null || r == false`?

Reply:
Cool, I put in `assert r != null` and removed the check. I also updated the javadoc to reflect.

Reply:
:+1: